### PR TITLE
refactor(storage): deprecate code of range delete

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Remove the code of range delete, only leave a `MonotonicDeleteEvent` for serde backward-compatibility. 

The `Serialize` and `Deserialize` requirement of `UserKey` and `TableKey` was derived from the serde requirement of `MonotonicDeleteEvent`. As we are deprecate `MonotonicDeleteEvent`, we can also deprecate the serde requirement on `UserKey` and `TableKey`. New replicated struct of `UserKey` and `TableKey` is added specially for the serde requirement of `MonotonicDeleteEvent`, so that the original `UserKey` and `TableKey` can get rid of the serde requirement.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/storage/src/hummock/sstable/mod.rs [hunk 1]
```diff
@@ -148,24 +78,14 @@ impl DeleteRangeTombstone {
 /// next event key wmk2 (7) (not inclusive).
 /// If there is no range deletes between current event key and next event key, `new_epoch` will be
 /// `HummockEpoch::MAX`.
-#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
 pub struct MonotonicDeleteEvent {
```
Comment (hzxa21) [inline src/storage/src/hummock/sstable/mod.rs:82]: Why do we still need to keep `MonotonicDeleteEvent` if we will never use it in the read path?

**Review by hzxa21 — APPROVED** [review]
LGTM. Thanks for efforts!

**Review by MrCroxx — APPROVED** [review]
LGTM

File: src/storage/src/hummock/sstable/mod.rs (hunk 1 above)
Comment (wenym1) [reply src/storage/src/hummock/sstable/mod.rs:82]: It's used for the backward compatibility to the previous encoding format of `SstableMeta`. If we are sure that all sst meta files written with this range delete event have been compacted in all users' cluster, we can remove this struct and change to return an error or panic when trying to decode it (seeing non-zero length in the list of this event).

**Pull request merged**